include/dmlc/optional.h

```diff
@@ -107,6 +108,8 @@ class optional {
   }
   /*! \brief whether this object is holding a value */
   explicit operator bool() const { return !is_none; }
+  /*! \brief alternate, more-explicit check of object value */
+  const bool IsNone() const { return is_none; }
```

Comment:
dmlc::optional is a temporary replacement for std:optional until we move to c++17.
The goal is to maintain consistency with std::optional interface.